src/engine/benchmark/source/hlp/hlp_bench.cpp

```diff
@@ -0,0 +1,384 @@
+#include <string>
+
+#include <benchmark/benchmark.h>
+
+// Including this private header directly to be able to
+// benchmark individual parsers.
+// Regular users of the library should NOT do this
+#include "src/SpecificParsers.hpp"
+#include <hlp/hlp.hpp>
+
+
+static std::string getRandomString(int len, bool includeSymbols = false) {
+    static const char alphanum[] =
+        "0123456789"
+        "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+        "abcdefghijklmnopqrstuvwxyz";
+
+    static const char symbols[] =
+        "-_'\\/. *!\"#$%&()+[]{},;";
+
+    std::string tmp_s;
+    tmp_s.reserve(len);
+
+    std::string dict = alphanum;
+    if(includeSymbols)
+    {
+        dict += symbols;
+    }
+
+    for (int i = 0; i < len; ++i) {
+        tmp_s += dict[rand() % dict.size()];
+    }
+    return tmp_s;
+}
+
+static std::string getRandomCapExprVariable(int len){
+    std::string ret;
+    for(int i = 0; i < len; ++i){
+        ret += '<';
+        ret += getRandomString((rand() % 10) + 1);
+        for(int j = 0; j < (rand() % 5); ++j){
+            ret += '/';
+            ret += getRandomString((rand() % 10) + 1);
+        }
+        ret += '>';
+        ret += getRandomString((rand() % 50) + 1, true);
+    }
+
+    return ret;
+}
+
+static std::string getRandomCapExpr(int len){
+    std::string ret;
+    for(int i = 0; i < len; ++i){
+        ret += '<';
+        ret += getRandomString(10);
+        for(int j = 0; j < 4; ++j){
+            ret += '/';
+            ret += getRandomString(10);
+        }
+        ret += '>';
+        ret += getRandomString(15, true);
+    }
+
+    return ret;
+}
+
+static std::string createRandomFilepath(int len, bool unixFormat = false, int length = 5)
+{
+    std::string ret, folder_separator;
+    if(unixFormat){
+        folder_separator = '/';
+        ret += getRandomString(len);
+    }
+    else {
+        folder_separator = '\\';
+        ret += "C:\\";
+    }
+    for(int i = 0; i < length; i++){
+        ret += getRandomString(len) + folder_separator ;
+    }
+
+    return ret;
+}
+
+static std::string createRandomDomain(int len, bool withSubdomain = false)
+{
+    std::string ret;
+    ret += "www.";
+    if(withSubdomain){
+        ret += getRandomString(len) + '.';
+    }
+    ret += getRandomString(len);
+    ret += ".com/";
+    ret += getRandomString(len);
+
+    return ret;
+}
+
+static std::string createMap(int len)
+{
+    std::string ret;
+    for(int i = 0; i < len; ++i){
+        ret += "key";
+        ret += std::to_string(i);
+        ret += '=';
+        ret += getRandomString(10);
+        if (i != (len - 1))
+            ret += ' ';
+    }
+    ret += ';';
+
+    return ret;
+}
+
+// TODO Probably need a way to mix-match a variable number of all our parsers
+static const char *logQL_expression = "<source.address> - <JSON> - [<timestamp/APACHE>] \"<http.request.method> "
```

Comment:
We should move this strings into the corresponding tests

Reply:
done